Only create directories and recompile when necessary

fholzhauser opened the pull request:
Hi,

Would you consider this pull request ? 
## Create working directories only when necessary

This is a cosmetic change. When using rebar3 to compile on release top level the provider also createdsthe working directories. With this change it only creates them if it has something to do (i.e. ASN.1 files found in the application)
## Adding noobj to options instead of deleting generated beam files

This is a cosmetic change. It does not result in any significant speedup.
## Avoid unconditional recompilation

ASN.1 generation is rather slow with more than a few files. During development of an application having ASN.1 files it is somewhat frustrating to have it with each recompilation unconditionally. With this change it compiles when any of the ASN.1 dictionaries have been updated (i.e. modification time is later than the corresponding generated .erl file).

Cheers,
Ferenc

fholzhauser committed:
Create working directories only when asn1 directory is present in app

fholzhauser committed:
Only recompile when any of the dictionaries have changed

knusbaum commented:
Hi! 
Thanks so much for this PR. I've been away from erlang for a while, so I'm only now looking at it.
This looks great except for the addition of noobj.

I'd prefer to keep default options as limited as possible so the plugin behaves as close to the plain asn1ct compile() function as possible, and since there's no particular practical benefit I would direct those who want that option to specify it in their build.

If you can remove that option, I will merge this.

fholzhauser committed:
Remove noobj default option

fholzhauser commented:
Hi,
I have removed noobj. Actually I think I've removed it earlier, as I've only found it in a debug output (which then probably was a bug as it should not have been there by now anyway).

knusbaum commented:
+1
Thanks for the contribution!

knusbaum commented:
New version 0.2.2 is up on hex.
